pack.go

```diff
@@ -151,19 +153,40 @@ func (p *Package) makeFilePackFunc(basePath string) filepath.WalkFunc {
 		defer r.Close()
 
 		// Copy data file to writer.
-		log.Printf("packer: writing '%s' to package", fi.Name())
-
-		if _, err = io.Copy(p, r); err != nil {
-			return err
+		log.Printf("writing '%s' to data package", fi.Name())
+
+		buf = make([]byte, 32*1024)
+
+		for {
+			nr, er := r.Read(buf)
+			if nr > 0 {
+				nw, ew := d.write(buf[0:nr])
+				if ew != nil {
+					err = ew
+					break
+				}
+				if nr != nw {
+					err = errors.New("short write")
+					break
+				}
+			}
+			if er == io.EOF {
+				break
+			}
+			if er != nil {
+				err = er
+				break
+			}
```

Comment:
It is a replacement of Copy to avoid exposing a Read function on DataPackage, which Copy requires but I thought would be confusing to users.